src/configurator/v1.ml

```diff
@@ -615,23 +612,32 @@ module Pkg_config = struct
         end
       | _ -> None
     in
-    if Process.run_process_ok c ~dir ?env t.pkg_config [expr] then
+    let pc_flags = "--print-errors" in
+    let { Process.exit_code; stderr; _ } =
+      Process.run_process c ~dir ?env t.pkg_config [pc_flags; expr] in
+    if exit_code = 0 then
       let run what =
         match String.trim (Process.run_capture_exn c ~dir ?env
                              t.pkg_config [what; package])
         with
         | "" -> []
         | s  -> String.split s ~on:' '
       in
-      Some
+      Ok
         { libs   = run "--libs"
         ; cflags = run "--cflags"
         }
     else
-      None
+      Error stderr
+
+  let res_forget = function Ok p -> Some p | Error _ -> None
```

Comment:
This is a standard function, so let's just add this to `Stdune.Result`. In base, this is called `Or_error.ok`, but that's probably too cute. Let's just call it `Result.to_option` or something.

Reply:
Indeed; I didn't add this to stdune I guess to keep it 1:1 to OCaml's result, but that's the wrong reason. Do you want me to update it on this PR or would you like it to introduce the change in a separate one?

Reply:
In stdune we extend the stdlib freely btw. We don't even bother to maintain compatibility. Dune_caml is the 1:1 "namespaced" clone, which we only need because `Stdlib` is >= 4.07. Feel free to just do it here, in a new PR, or just push to master. This is a very a minor change.

Reply:
Ok will do it here as this PR is the first consumer of such a function.